tls/s2n_resume.c

```diff
@@ -769,187 +769,147 @@ struct s2n_ticket_key *s2n_find_ticket_key(struct s2n_config *config, const uint
     return NULL;
 }
 
-int s2n_encrypt_session_ticket(struct s2n_connection *conn, struct s2n_stuffer *to)
+S2N_RESULT s2n_resume_encrypt_session_ticket(struct s2n_connection *conn, struct s2n_stuffer *to)
 {
-    struct s2n_ticket_key *key = NULL;
-    struct s2n_session_key aes_ticket_key = { 0 };
-    struct s2n_blob aes_key_blob = { 0 };
-
-    uint8_t iv_data[S2N_TLS_GCM_IV_LEN] = { 0 };
-    struct s2n_blob iv = { 0 };
-    POSIX_GUARD(s2n_blob_init(&iv, iv_data, sizeof(iv_data)));
-
-    uint8_t aad_data[S2N_TICKET_AAD_LEN] = { 0 };
-    struct s2n_blob aad_blob = { 0 };
-    POSIX_GUARD(s2n_blob_init(&aad_blob, aad_data, sizeof(aad_data)));
-    struct s2n_stuffer aad = { 0 };
+    RESULT_ENSURE_REF(conn);
+    RESULT_ENSURE_REF(to);
 
+    struct s2n_ticket_key *key = NULL;
     key = s2n_get_ticket_encrypt_decrypt_key(conn->config);
 
     /* No keys loaded by the user or the keys are either in decrypt-only or expired state */
-    S2N_ERROR_IF(!key, S2N_ERR_NO_TICKET_ENCRYPT_DECRYPT_KEY);
+    RESULT_ENSURE(key != NULL, S2N_ERR_NO_TICKET_ENCRYPT_DECRYPT_KEY);
```

Comment:
Nit: I'd get rid of the newline here. I was like "where do we check that key != NULL after setting it" before realizing it happens one "paragraph" later. And do you need to initially set key to NULL? Doesn't really look like it helps with readability / line length or something.